### PR TITLE
Check that certain decorations cannot be used more than once and/or are mutually exclusive

### LINKED ISSUE #1636: validator: Certain decorations should be used at most once on an ID
Example: Layout decorations such as Offset, ArrayStride, RowMajor, ColMajor

In fact RowMajor and ColMajor are mutually exclusive.

### PR DESCRIPTION
**Pull request by ale64bit**
Fixes #1636

### CODE REVIEW

**Comment by ale64bit** [comment]
PTAL

File: source/val/validate_decorations.cpp [hunk 1]
```diff
@@ -917,6 +926,86 @@ spv_result_t CheckDecorationsOfBuffers(ValidationState_t& vstate) {
...
+  static const std::unordered_set<SpvDecoration, SpvDecorationHash>
+      kAtMostOncePerMember = {
+          SpvDecorationOffset,
+          SpvDecorationMatrixStride,
+          SpvDecorationRowMajor,
+          SpvDecorationColMajor,
+      };
+  static const std::vector<std::unordered_set<SpvDecoration, SpvDecorationHash>>
+      kMutuallyExclusivePerMember = {
+          {SpvDecorationRowMajor, SpvDecorationColMajor},
```
Comment (alan-baker) [inline source/val/validate_decorations.cpp]: I would add BufferBlock and Block here too although they are not member decorations.

File: source/val/validate_decorations.cpp [hunk 2]
```diff
@@ -917,6 +926,86 @@ spv_result_t CheckDecorationsOfBuffers(ValidationState_t& vstate) {
   return SPV_SUCCESS;
 }
 
+spv_result_t CheckDecorationsCompatibility(ValidationState_t& vstate) {
+  static const std::unordered_set<SpvDecoration, SpvDecorationHash>
```
[the remark this answers is not in the document]
Comment (ale64bit) [reply source/val/validate_decorations.cpp]: Changed to unique_ptr. Not sure if that's what you intended, but PTAL.

File: source/val/validate_decorations.cpp (hunk 1 above)
Comment (ale64bit) [reply source/val/validate_decorations.cpp]: Done

**Comment by ale64bit** [comment]
PTAL

File: source/val/validate_decorations.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ale64bit) [reply source/val/validate_decorations.cpp]: Changed to "static const auto* const impl = new T(args...);" after re-reading style guide.

**Comment by alan-baker** [comment]
Please rebase and fix the formatting.

**Pull request merged**